showkase-processor/src/main/java/com/airbnb/android/showkase/processor/writer/ShowkaseBrowserWriter.kt

```diff
@@ -144,6 +159,48 @@ internal class ShowkaseBrowserWriter(private val environment: XProcessingEnv) {
         .addMember("numTypography = %L", typographySize)
         .build()
 
+    // This is to aggregate metadata for the custom annotation annotated with Preview
+    internal fun writeCustomAnnotationElementToMetadata(element: XElement) {
+        if (!element.isTypeElement()) return
+        if (element.isAnnotationClass() && element.qualifiedName == ShowkaseProcessor.PREVIEW_CLASS_NAME) return
+
+        val moduleName = "Showkase_${element.qualifiedName.replace(".", "_")}"
+        val generatedClassName =
+            "ShowkaseMetadata_${moduleName.lowercase(Locale.getDefault())}"
+
+        val previewAnnotations =
+            element.getAllAnnotations().filter { it.name == ShowkaseProcessor.PREVIEW_SIMPLE_NAME }
+
+        val fileBuilder = FileSpec.builder(
+            ShowkaseProcessor.CODEGEN_PACKAGE_NAME,
+            generatedClassName
+        )
+
+        val functions = previewAnnotations.mapIndexed { index, xAnnotation ->
+            FunSpec.builder("${xAnnotation.name}_$index")
+                .addAnnotation(
+                    AnnotationSpec
+                        .builder(ShowkaseMultiPreviewCodegenMetadata::class)
+                        .addMember("previewName = %S", xAnnotation.get("name"))
+                        .addMember("previewGroup = %S", xAnnotation.get("group"))
+                        .addMember("supportTypeQualifiedName = %S", element.qualifiedName)
+                        .addMember("showkaseWidth = %L", xAnnotation.getAsInt("widthDp"))
+                        .addMember("showkaseHeight = %L", xAnnotation.getAsInt("heightDp"))
+                        .addMember("packageName = %S", element.packageName)
+                        .build()
+                ).build()
+        }
+
+        fileBuilder.addType(
+            TypeSpec.classBuilder(generatedClassName).addFunctions(functions).build()
+        ).addFileComment("This is an auto-generated file. Please do not edit/modify this file.")
+        try {
+            fileBuilder.build().writeTo(environment.filer, mode = XFiler.Mode.Aggregating)
+        } catch (fileExists: FileAlreadyExistsException) {
```

Comment:
If you remove this, are you seeing an error/crash? I wonder if this is masking a potential actual issue in the implementation 🤔

Reply:
I removed it and tested locally and did not see any difference, so removed it in 855cd00 :) Thanks for pointing it out! :)